xchange-bitfinex/src/test/java/info/bitrich/xchangestream/bitfinex/BitfinexManualAuthExample.java

```diff
@@ -0,0 +1,38 @@
+package info.bitrich.xchangestream.bitfinex;
+
+import info.bitrich.xchangestream.core.StreamingExchangeFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BitfinexManualAuthExample {
+    private static final Logger LOG = LoggerFactory.getLogger(BitfinexManualAuthExample.class);
+
+    public static void main(String[] args) {
+        BitfinexStreamingExchange exchange = (BitfinexStreamingExchange) StreamingExchangeFactory.INSTANCE.createExchange(
+            BitfinexStreamingExchange.class.getName());
+        exchange.setCredentials("API-KEY", "API-SECRET");
+
+        exchange.connectToAuthenticated().blockingAwait();
+        exchange.getStreamingAuthenticatedDataService().getAuthenticatedTrades().subscribe(
```

Comment:
SO, to clarify my earlier comments, this would be more "standard" like this:

```
ExchangeSpecification specification = StreamingExchangeFactory.INSTANCE.createExchange(BitfinexStreamingExchange.class.getName()).getDefaultExchangeSpecification();
specification.setApiKey("API-KEY");
specification.setSecretKey("API-SECRET");
BitfinexStreamingExchange exchange = (BitfinexStreamingExchange)StreamingExchangeFactory.INSTANCE.createExchange(specification);

exchange.getStreamingMarketDataService().getRawAuthenticatedTrades().subscribe(
 ...
);
```

So, use the existing authentication details provided by the exchange specification, authenticate by default if the authentication details are available, and use L1 "raw" methods on the `StreamingMarketDataService` itself.

Reply:
Done on my uplift fork